### ISSUE TITLE
pageSize opts not working

### ISSUE THREAD

**Issue by simonemazzoni** [twilio/twilio-node#448]
Hello, I'm facing a wrong behaviour using the twilio-node sdk. If I try to retrieve the messages list from a channel setting a pageSize value, let's say equal to 25, the result contains all the messages like if the pageSize option is ignored.

I tried using the REST api instead, specifying the `?PageSize=25` param, and all works fine since I get the first 25 messages and the link for the next page and so on.

How can I solve this? Thanks.
Is there also a way to specify the starting page?

Thanks,
Simone

**Version:**
3.30.3
### Code Snippet
```node
const twilio = require('twilio');

const accountSid = 'acc sid';
const apikey_sid = 'apikey sid';
const apikey_secret = 'apikey secret';

const client = new twilio.Twilio(apikey_sid, apikey_secret, { accountSid: accountSid });

client.chat
    .services('serviceSid')
    .channels(encodeURIComponent('channelSid'))
    .messages
    .list({pageSize: 25})
        .then(ch => console.log("CHANNEL:", ch))
        .catch(err => console.error("ERR:", err));
```

**Comment by childish-sambino** [comment]
Use the `limit` option instead.

**Comment by simonemazzoni** [comment]
@childish-sambino I don't think it is the same, I need to paginate the results, not to limit them.

**Comment by childish-sambino** [comment]
Ah, sorry. I misread. Try something like this:
```
try {
    const messages = client.chat
        .services('serviceSid')
        .channels(encodeURIComponent('channelSid'))
        .messages;

    const page0 = await messages.page({ pageSize: 25 });
    page0.instances.forEach(message => console.log("MESSAGE:", message));

    // Get the next page.
    const page1 = await messages.getPage(page0.nextPageUrl);
    page1.instances.forEach(message => console.log("MESSAGE:", message));

} catch (err) {
    console.error("ERR:", err);
}
```

or

```
try {
    const page0 = await client.chat
        .services('serviceSid')
        .channels(encodeURIComponent('channelSid'))
        .messages
        .page({ pageSize: 25 });

    page0.instances.forEach(message => console.log("MESSAGE:", message));

    // Get the next page.
    const page1 = await page0.nextPage();
    page1.instances.forEach(message => console.log("MESSAGE:", message));

} catch (err) {
    console.error("ERR:", err);
}
```

**Comment by simonemazzoni** [comment]
Hi @childish-sambino, I tried your suggestions and it works.
I noticed that the `page()` function has 4 options params, `pageSize` `pageNumber` `order` and `pageToken`. What is exactly the purpose of `pageToken`?

Also, I have a lot of difficulties finding the documentation of the twilio-node package. Is there a dedicated page? Where I can find it?

thanks

**Comment by childish-sambino** [comment]
There's a link in the project description which contains links for the generated docs for each release: http://twilio.github.io/twilio-node/

Latest release for the function in question: https://www.twilio.com/docs/libraries/reference/twilio-node/3.31.0/Twilio.Chat.V1.ServiceContext.ChannelContext.MessageList.html#page

There's some info here regarding pagination: https://www.twilio.com/docs/usage/twilios-response?code-sample=code-paging-information-4

I _think_ `pageToken` is similar to a page number in that it indicates a starting point to retrieve results. It's included in `nextPageUrl` as part of the returned `page`.

**Comment by simonemazzoni** [comment]
@childish-sambino thanks for the answer, that documentation is more clear. I think it would be nice to have a link for that on the readme of the library itself because it is very difficult to find. 😄 

Also, if I'm not wrong, it seems that `pageNumber` does nothing, but it is only a param meant to be used on the client to keep the count of requested pages. I see that if I change the `pageNumber` but I don't change the `pageToken` between subsequent requests the result is always the same.

Is it correct?

**Comment by childish-sambino** [comment]
Yes, that's correct.

**Comment by waldothedeveloper** [comment]
> Ah, sorry. I misread. Try something like this:
> 
> ```
> try {
>     const messages = client.chat
>         .services('serviceSid')
>         .channels(encodeURIComponent('channelSid'))
>         .messages;
> 
>     const page0 = await messages.page({ pageSize: 25 });
>     page0.instances.forEach(message => console.log("MESSAGE:", message));
> 
>     // Get the next page.
>     const page1 = await messages.getPage(page0.nextPageUrl);
>     page1.instances.forEach(message => console.log("MESSAGE:", message));
> 
> } catch (err) {
>     console.error("ERR:", err);
> }
> ```
> 
> or
> 
> ```
> try {
>     const page0 = await client.chat
>         .services('serviceSid')
>         .channels(encodeURIComponent('channelSid'))
>         .messages
>         .page({ pageSize: 25 });
> 
>     page0.instances.forEach(message => console.log("MESSAGE:", message));
> 
>     // Get the next page.
>     const page1 = await page0.nextPage();
>     page1.instances.forEach(message => console.log("MESSAGE:", message));
> 
> } catch (err) {
>     console.error("ERR:", err);
> }
> ```

I wish examples like this would be part of Twilio's documentation.
